respy/shared.py

```diff
@@ -411,3 +424,103 @@ def create_state_space_columns(optim_paras):
     return create_core_state_space_columns(
         optim_paras
     ) + create_dense_state_space_columns(optim_paras)
+
+
+def apply_to_state_space_attribute(attribute, func):
+    """Apply a function to a state space attribute which might be dense or not."""
```

Comment:
The docstring would be better with an example for a StateSpace attribute. What type are StateSpace attributes? Dictionaries?